### DISCUSSION TITLE
Protocol Buffers not listing on repo

### DISCUSSION THREAD

**Discussion by psybers** [github-linguist/linguist#6378] [Classification] [answered]
For one of my repos (https://github.com/boalang/compiler) I noticed the detected languages does not list Protocol Buffers.  It appears there is support for detected protobuf in Linguist:

```
src/proto/ast.proto: 560 lines (542 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/code.proto: 71 lines (67 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/control.proto: 51 lines (46 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/diff.proto: 125 lines (114 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/graph.proto: 21 lines (19 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/issues.proto: 158 lines (149 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/shared.proto: 51 lines (38 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
src/proto/toplevel.proto: 83 lines (80 sloc)
  type:      Text
  mime type: text/plain
  language:  Protocol Buffer
```

But when I run on the whole repository, it does not show any `Protocol Buffer` results:

```
97.55%  2267410    Java
2.12%   49201      ANTLR
0.10%   2239       JavaScript
0.08%   1909       PHP
0.06%   1441       HTML
0.04%   997        CSS
0.03%   694        Batchfile
0.02%   520        Shell
```

Shouldn't this list the files above?

**Comment by lildude** [comment, marked as answer]
> Shouldn't this list the files above?

Nope. Protobuf files are considered data:

https://github.com/github/linguist/blob/5a0c74277548122267d84283910abd5e0b89380e/lib/linguist/languages.yml#L5282-L5293

... so are not considered by default (from "[How Linguist works](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md)"):

> Linguist starts by going through all the files in a repository and excludes all files that it determines to be binary data, [vendored code](/docs/overrides.md#vendored-code), [generated code](/docs/overrides.md#generated-code), [documentation](/docs/overrides.md#documentation), or are defined as `data` (e.g. SQL) or `prose` (e.g. Markdown) languages, whilst taking into account any [overrides](/docs/overrides.md).

If you want them to count, you'll need to use the [`linguist-detectable` override](https://github.com/github/linguist/blob/master/docs/overrides.md#detectable).

**Comment by psybers** [comment]
Great, thank you for the pointer to the overrides docs. I didn't spot that before!  That fixed my specific case (and other issues!).
